GDAS-validation/setup_workspace.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+# setup_workspace.sh
+# this script:
+# - creates working directory
+# - clones global-workflow
+# - clones GDASApp
+# - clones GSI
+# - builds GDASApp + GSI
+# - creates an experiment for a specified cycle/period
+# - links input background files
+
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 [-c] [-b] [-s]"
+  echo
+  echo "  -c  clone necessary repositories"
+  echo "  -b  build GDASApp and GSI"
+  echo "  -s  setup default experiment"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+clone=NO
+build=NO
+setup=NO
+
+while getopts "cbsh" opt; do
+  case $opt in
+    c)
+      clone=YES
+      ;;
+    b)
+      build=YES
+      ;;
+    s)
+      setup=YES
+      ;;
+    h|\?|:)
+      usage
+      ;;
+  esac
+done
+
+#--------------- User modified options below -----------------
```

Comment:
Good to section off the variables which users can or may want to modify.   A few suggestions

1. Consider reducing the _user change_ section to only the variables which users can set.  Move actions based on user set variables to the next (_don't change_)  section.  For example, the _user change_ section could be trimmed down to
```
EXPNAME="gdas_eval_satwnd"
machine="orion"
workdir=""
ICSDir=""
```
`EXPNAME` is the variable users will alter the most.  

2. Consider using GDASApp or g-w `ush/detect_machine.sh` to set `machine` in the _don't change_ section.   Of course, it's perfectly fine for a user to manually set `machine="orion"`.

3. `wrkdir` won't be changed as often since a given `wrkdir` can support multiple `EXPNAME`.  It's unlikely `ICSDir` will be changed since most developers won't stage their own initial conditions, will they?   We could move `ICSDir` to the _don't change_ section.  The same could be said for `wrkdir`.   I set `wrkdir` and `ICSDir` to "" above with the intention that defaults provided in the _don't change_ section are used.

Reply:
I like the idea of using `detect_machine.sh` but we have a chicken and egg problem as it is not yet there to use until the working directory is created and things are cloned in it. Should I duplicate that script in this PR?

Reply:
I'm working on refactoring the user modifiable section now

Reply:
`setup_workspace.sh` uses `machine` in two places.   The second usage is in the `-s` block.   Since `-b` precedes this step we already have a cloned copy of `detect_machine.sh` in g-w and GDASApp.   The first usage of `machine` is problematic if we want to use `detect_machine.sh`.   While we can move the definition of `ICSDir` to the `-s` block, we need to define `workdir` at the start of the `-b` section.  This is before we clone g-w or GDASApp.

We can simply leave `machine` as a user set variable.

Reply:
e4f3680 includes all suggested changes except for the `detect_machine.sh` inclusion